### PR TITLE
Enable full-screen viewing of images upon click

### LINKED ISSUE #1245: Feature Request: see images in full screen on click
Currently when looking at plugin doc, the images can be too small to be readable, and when we click on them we are redirected to GitHub.
Example: https://plugins.jenkins.io/config-file-provider/

I'd like a modal (or similar) to be open on click instead, displaying the image filling the screen and dismissed with another click.

### PR DESCRIPTION
**Pull request by sridamul**
Fixes #1245

Specifically targeting the `` tag within the `<a target="_blank">` tag, where all the PluginWiki images are indexed.

Demo:

https://github.com/jenkins-infra/plugin-site/assets/107404972/2917a062-4219-409e-8f27-ba0d8e4c971e

### CODE REVIEW

File: plugins/plugin-site/src/components/PluginWikiContent.jsx [hunk 1]
```diff
@@ -1,9 +1,102 @@
...
+        }
+    };
+
     if (wiki?.childHtmlRehype) {
-        return <div className="content" dangerouslySetInnerHTML={{__html: wiki.childHtmlRehype.html}} />;
+        const {html} = wiki.childHtmlRehype;
+
+        return (
+            <div className="content" onClick={handleClick} style={getStyle()}>
+                <style>
```
Comment (halkeye) [inline plugins/plugin-site/src/components/PluginWikiContent.jsx]: can you move the styles to a css file? preferably the .module.css, we try not to inline css for this project

File: plugins/plugin-site/src/components/PluginWikiContent.jsx [hunk 2]
```diff
@@ -1,9 +1,102 @@
...
+            parentNode.tagName === 'A' &&
+            parentNode.getAttribute('target') === '_blank'
+        ) {
+            // Prevent the default behavior of the anchor tag (prevents redirection to github page)
+            event.preventDefault();
+            const imgElement = parentNode.querySelector('img');
+            if (imgElement) {
+                if (imgElement.id === 'imageResize') {
+                    imgElement.removeAttribute('id');
+                    setIsClicked(false);
```
Comment (halkeye) [inline plugins/plugin-site/src/components/PluginWikiContent.jsx]: how is something not clicked inside the click handler.

Maybe this should be is opened? is lightbox toggled, etc

File: plugins/plugin-site/src/components/PluginWikiContent.jsx [hunk 3]
```diff
@@ -1,9 +1,102 @@
...
+                        left: 0;
+                        width: 100%;
+                        height: 100%;
+                        background-color: rgba(0, 0, 0, 0.5);
+                        z-index: 9998;
+                    }
+                `}
+                </style>
+                <div dangerouslySetInnerHTML={{__html: html}} />
+                {isClicked && document.getElementById('imageResize') && (
```
Comment (halkeye) [inline plugins/plugin-site/src/components/PluginWikiContent.jsx]: why do we need to check if a specific element is in the dom? shouldn't we know if it is there or not?

File: plugins/plugin-site/src/components/PluginWikiContent.jsx (hunk 2 above)
Comment (sridamul) [reply plugins/plugin-site/src/components/PluginWikiContent.jsx]: @halkeye I renamed `isClicked` to `isLightBoxOpen`. Is this Ok?

**Comment by sridamul** [comment]
@halkeye Is this a best practice? 
```
const resetImage = () => {
        const images = document.getElementsByClassName(imageResize);
        for (let i = 0; i < images.length; i++) {
            const image = images[i];
            image.removeAttribute('class');
        }
        setIsLightboxOpen(false);
    };
```
I thought about removing the specific class with the className instead of removing the whole class attribute (tested in local env - working). 

If neither is a best practice, please show me the right path.

**Comment by sridamul** [comment]
@zbynek Can you review this?

**Comment by halkeye** [comment]
please give people time, everyone here has a personal life and other responsibilities.

Its been years since I did any react work, but it feels like document.* functions seem wrong, those feel like hacks. React should manage its own dom, and in the rare case it doesn't, it should use something like useRef no?

**Comment by sridamul** [comment]
> please give people time, everyone here has a personal life and other responsibilities.

Sorry for the inconvenience. Won't happen again

**Comment by sridamul** [comment]
@halkeye,I am not sure about this conversion from direct DOM manipulation to useRef implementation. Could you help me when you have time?

Also, I kindly apologize for asking for the review multiple times.

File: plugins/plugin-site/src/components/PluginWikiContent.jsx [hunk 4]
```diff
@@ -1,9 +1,46 @@
...
+        while (parentNode && parentNode.tagName !== 'A') {
+            parentNode = parentNode.parentNode;
+        }
+
+        if (
+            parentNode &&
+            parentNode.tagName === 'A' &&
+            parentNode.getAttribute('target') === '_blank'
+        ) {
+            // Prevent the default behavior of the anchor tag (prevents redirection to github page)
```
Comment (krisstern) [inline plugins/plugin-site/src/components/PluginWikiContent.jsx]:
```suggestion
            // Prevent the default behavior of the anchor tag from redirecting to the GitHub page
```

Comment (krisstern) [reply plugins/plugin-site/src/components/PluginWikiContent.jsx]: @sridamul Is this what you originally meant? Would it be better to rephrase for clarity?

Comment (sridamul) [reply plugins/plugin-site/src/components/PluginWikiContent.jsx]: >  Is this what you originally meant? 

Yes

**Review by NotMyFault — APPROVED** [review]
Looks good, thanks!

**Comment by sridamul** [comment]
@MarkEWaite Is it ready to ship?

Also May I know why the CI fails?

**Comment by MarkEWaite** [comment]
Someone with access to infra.ci.jenkins.io will need to investigate.  I've not reviewed the pull request, so I am not able to answer if it is ready to merge

**Comment by sridamul** [comment]
Could you please tell me why the CI is failing, @NotMyFault?

**Comment by krisstern** [comment]
@sridamul Please if possible do not ping individuals to follow up on your PR, you can notify the repo maintainers without doing so.

**Comment by NotMyFault** [comment]
The CI build is now green.

**Comment by sridamul** [comment]
Is it ready to ship or require any changes?

**Pull request merged**